File: src/main/java/com/tms/csp/transforms/RmpsToCnf.java
package com.tms.csp.transforms;

import com.tms.csp.ast.Exp;

/**
 * Replaces each iff(a b) with an and(imp(a b) imp(b a))
 */
public class RmpsToCnf extends BaseTransformer {

    @Override
    protected Exp local(Exp in) {
        return in.asRmp().toOr();
    }

    @Override
    protected boolean executeLocal(Exp in) {
        return in.isRmp();
    }


}

File: src/main/java/com/tms/csp/transforms/RmpsToCnf.kt
package com.tms.csp.transforms

import com.tms.csp.ast.Exp

/**
 * Replaces each iff(a b) with an and(imp(a b) imp(b a))
 */
class RmpsToCnf : BaseTransformer() {

    override fun local(inExp: Exp): Exp? {
        return inExp.asRmp().toOr
    }

    override fun executeLocal(inExp: Exp): Boolean {
        return inExp.isRmp
    }


}
